src/core/border_router/routing_manager.cpp

```diff
@@ -514,6 +525,11 @@ void RoutingManager::UnpublishLocalOmrPrefix(void)
     }
 }
 
+bool RoutingManager::IsOmrPrefixAddedToLocalNetworkData(void) const
+{
+    return Get<NetworkData::Local>().ContainsOnMeshPrefix(mLocalOmrPrefix);
```

Comment:
@simonlingoogle , can you please help add a short description on what is the election issue  that this PR fixes. I am not sure if I can follow the issue from the code.Thanks.

One thing that I was considering related to this change (that may come into focus in future), here the check is in the local network data (i.e. expecting that OMR is added by this device itself), but at some point we may use `NetworkData::Publisher` to limit number of similar OMR prefixes added into the Thread Network Data. In such a case the prefix entry may not be added in the local network data immediately (even if is already published by this device). I guess doing the check in `Local` network data may not work then. How about adding a flag or state variable in `RoutingManager` itself to track whether or not we have published a prefix in Thread network data (use the flag then instead of `IsOmrPrefixAddedToLocalNetworkData()`. Thoughts?

Reply:
> at some point we may use `NetworkData::Publisher` to limit number of similar OMR prefixes added into the Thread Network Data

Currently, there is only one OMR prefix added by one BR when stabilized. So, we may not need to use Publisher for OMR Prefix in the near future?

Reply:
> can you please help add a short description on what is the election issue that this PR fixes. I am not sure if I can follow the issue from the code.Thanks.

I will elabrate on the code.

Reply:
> Currently, there is only one OMR prefix added by one BR when stabilized. So, we may not need to use Publisher for OMR Prefix in the near future?

I suggest we change change this (for redundancy) in future.

But independent of this, would it be much simpler to add a `bool` flag in `RoutingManger` to track whether or not prefix  is already added or not?
- Makes the code and logic simpler (easier tor read), 
- We don't need to add new methods (`ContainsOnMeshPrefix`) in `NetworkData::Local` and,
- The code will work fine when we move to use Publisher.

Reply:
Adding redundancy for the OMR prefix has its challenges. The redundancy implies that BRs will advertise an OMR prefix that they do not own. When the owner of the OMR prefix leaves the partition, it is difficult to know whether the owner is advertising that OMR prefix in another partition.

Reply:
> When the owner of the OMR prefix leaves the partition, it is difficult to know whether the owner is advertising that OMR prefix in another partition.

+1.

I remember there are thoughts of using mDNS to advertise OMR prefixes which are being used in the Thread partition so that a BR can stop advertise the same OMR prefix when it's discovered via mDNS. https://threadgroup.atlassian.net/browse/SPEC-984 looks like is adding OMR prefixes into the `_meshcop._udp` service.

Reply:
> But independent of this, would it be much simpler to add a `bool` flag in `RoutingManger` to track whether or not prefix is already added or not?

Adding a `bool` may add more code size and ram footprint. Thus prefer the current approach slightly.

Reply:
> > When the owner of the OMR prefix leaves the partition, it is difficult to know whether the owner is advertising that OMR prefix in another partition.
> 
> +1.
> 
> I remember there are thoughts of using mDNS to advertise OMR prefixes which are being used in the Thread partition so that a BR can stop advertise the same OMR prefix when it's discovered via mDNS. https://threadgroup.atlassian.net/browse/SPEC-984 looks like is adding OMR prefixes into the `_meshcop._udp` service.

I think it can work. But it's hard to judge whether the benefit is worth the complexity.

Reply:
It should be straightforward to refactor it to using bool flag in the future. 
I will keep it as is for now.